<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MotorWidget</class>
 <widget class="QWidget" name="MotorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>963</width>
    <height>700</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>780</width>
    <height>550</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1900</width>
    <height>700</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MotorWidget</string>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>8</y>
     <width>151</width>
     <height>561</height>
    </rect>
   </property>
   <property name="title">
    <string>Motor list</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_11">
    <item>
     <widget class="QTreeWidget" name="treeWidget_card">
      <property name="contextMenuPolicy">
       <enum>Qt::CustomContextMenu</enum>
      </property>
      <column>
       <property name="text">
        <string notr="true">1</string>
       </property>
      </column>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>172</x>
     <y>14</y>
     <width>541</width>
     <height>561</height>
    </rect>
   </property>
   <property name="title">
    <string/>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_13">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_12" stretch="0,1,3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label_28">
            <property name="text">
             <string>Axis name</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEdit_axisName"/>
          </item>
          <item>
           <spacer name="horizontalSpacer_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_disableAxis">
          <property name="text">
           <string>ban</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout" stretch="3,1,3">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_24" stretch="2,3">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Motor type</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_motortype"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_25" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_23">
               <property name="text">
                <string>Enable limit</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_limits_mode"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_26" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_24">
               <property name="text">
                <string>Enable Alarm</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_alarm_mode"/>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_27" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_20">
               <property name="text">
                <string>Home sense</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_home_sense"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_28" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Limit sense</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_limits_sense"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_29" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_25">
               <property name="text">
                <string>Alarm sense</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_alarm_sense"/>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,1">
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>Work Parameter</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Step value</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_stepvalue"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Negative soft limit</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_limitN"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_12" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>positive soft limit</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_limitP"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>Speed</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_vel"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_14" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_12">
               <property name="text">
                <string>Work acc time</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_accTime"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_15" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>Work dec time</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_decTime"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_16" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_30">
               <property name="text">
                <string>Stop stime</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_stopTime"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_17" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_29">
               <property name="text">
                <string>Stop mode</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_stopMode"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>Home Parameter</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>Home dir</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_home_dir"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Home type</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_home_mode"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_18" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>search distance</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_sreachhomePos"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_19" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_14">
               <property name="text">
                <string>Offset</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_inHomeOffset"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_20" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>After home offset</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_afterhomeOffset"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_21" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Home speed</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_homeVel"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_22" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_26">
               <property name="text">
                <string>Home acc time</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_homeAccTime"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_23" stretch="2,3">
             <item>
              <widget class="QLabel" name="label_27">
               <property name="text">
                <string>Home dec time</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit_homeDecTime"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="groupBox_3">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>8</y>
     <width>179</width>
     <height>561</height>
    </rect>
   </property>
   <property name="title">
    <string>Axis Debug</string>
   </property>
   <widget class="QPushButton" name="pushButton_home">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>120</y>
      <width>103</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>home</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>260</y>
      <width>163</width>
      <height>287</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="3,2">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>5</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QLabel" name="label_22">
         <property name="text">
          <string>target pos</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>current pos</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>servo alarm</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>Origin</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>Positive Limit</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Negative Limit</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string> Position signal</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="label_pos_to">
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_pos_real">
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_warn">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_home">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_plimits">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_nlimits">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_inplace">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton_stop">
    <property name="geometry">
     <rect>
      <x>94</x>
      <y>178</y>
      <width>57</width>
      <height>29</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 0, 0);</string>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_step">
    <property name="geometry">
     <rect>
      <x>66</x>
      <y>28</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_Pmove">
    <property name="geometry">
     <rect>
      <x>18</x>
      <y>30</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Jog</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_axisInfo_2">
    <property name="geometry">
     <rect>
      <x>138</x>
      <y>28</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>mm</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_enable">
    <property name="geometry">
     <rect>
      <x>34</x>
      <y>186</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Enable</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>19</x>
      <y>66</y>
      <width>143</width>
      <height>33</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QPushButton" name="pushButton_moven">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>60</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_movep">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>60</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>+</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>4</x>
     <y>620</y>
     <width>847</width>
     <height>32</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="pushButton_save">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>75</width>
        <height>30</height>
       </size>
      </property>
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="pushButton_cancel">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>75</width>
        <height>30</height>
       </size>
      </property>
      <property name="text">
       <string>Cancel</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_cancel</sender>
   <signal>clicked()</signal>
   <receiver>MotorWidget</receiver>
   <slot>onCancel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>395</x>
     <y>405</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>397</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_save</sender>
   <signal>clicked()</signal>
   <receiver>MotorWidget</receiver>
   <slot>onSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>472</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>512</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onCancel()</slot>
  <slot>onSave()</slot>
  <slot>onEdit()</slot>
  <slot>onHome()</slot>
  <slot>onMoveN()</slot>
  <slot>onMoveP()</slot>
 </slots>
</ui>
